reanalysis/utils.py

```diff
@@ -0,0 +1,342 @@
+"""
+a collection of classes and methods
+which may be shared across reanalysis components
+"""
+
+
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from dataclasses import dataclass, is_dataclass
+from csv import DictReader
+import json
+import re
+import sys
+
+import cyvcf2
+from cloudpathlib import AnyPath
+from cyvcf2 import Variant
+
+# modify how to import cache based on the system python
+v_info = sys.version_info
+if v_info.major != 3:
+    raise Exception('Come on, use Python3')
+if v_info.minor < 9:
+    from functools import lru_cache as cache
+else:
+    from functools import cache
+
+# sample: gene: variant: [partner_variant, ...]
```

Comment:
Might be worth explaining the string format of each of these (or potentially adding more detail below for `VARIANT_STRING_TEMPLATE` etc).